lib/icingadb/redisconnection.cpp

```diff
@@ -338,18 +338,15 @@ void RedisConnection::WriteLoop(asio::yield_context& yc)
 	for (;;) {
 		m_QueuedWrites.Wait(yc);
 
-	WriteFirstOfHighestPrio:
-		for (auto& queue : m_Queues.Writes) {
-			if (m_SuppressedQueryKinds.find(queue.first) != m_SuppressedQueryKinds.end() || queue.second.empty()) {
+		while (!m_Queues.Writes.empty()) {
+			if (m_SuppressedQueryKinds.find(m_Queues.Writes.top().Priority) != m_SuppressedQueryKinds.end()) {
```

Comment:
Please explain (and don't change anything regarding this, yet):

What is and was this condition for?
What happens now (compared to the previous behavior) if it's true?

Reply:
Before that, after each loop iteration, i.e. when the top priority has been successfully written and removed from the queue, you would have to start again from the beginning so that the top priority is always written first. But now we don't have to do this, because the highest priority is always at the top of the queue, when the first highest priority is successfully written and removed, the second highest priority is now at the top, and so on...

And the if condition is the same as before only now we don't also check for an empty queue content because we do this with the While loop.

Reply:
If I have to think about it, we don't need this If condition at all, because the priorities which are already in `WriteQueueItem` can never be in `m_SuppressedQueryKinds`.

Reply:
Of course they can!

Reply:
Have seen it!